nvtabular/ops/base.py

```diff
@@ -103,7 +102,43 @@ def supports(self) -> Supports:
         """ Returns what kind of data representation this operator supports """
         return Supports.CPU_DATAFRAME | Supports.GPU_DATAFRAME
 
-    def inference_initialize(self, columns: ColumnNames, model_config: dict) -> Optional[Operator]:
+    def inference_initialize(self, columns: Columns, model_config: dict) -> Optional[Operator]:
         """Configures this operator for use in inference. May return a different operator to use
         instead of the one configured for use during training"""
         return None
+
+
+class AddMetadata(Operator):
```

Comment:
Where/how does this operator get used?

Reply:
I haven't ported this yet, but an example can be found [here](https://github.com/NVIDIA/NVTabular/blob/romeyn/input-api/nvtabular/dataset/movielens.py#L35). This allows for manual tagging of columns. I should add some tests showcasing this as well.

Reply:
Ahh, if this is used in the Datasets API, it might fit better in that PR.

Reply:
I would argue that tagging columns manually is a core part of the API proposed in this PR. I do agree with @jperez999 that we should break this out into a separate file.

Reply:
Hmm, okay. It's a little hard to tell from the content of the PR itself without examples/tests, but I can get on board with manual tagging as important part of this API. Agreed with y'all that we should break it out then.